msocasecharge.yaml

```diff
@@ -0,0 +1,183 @@
+entityDefinitions:
+
+  person:
+    fqn: "nc.SubjectIdentification"
+    entitySetName: "MSOInmates"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
+        column: "PCP"
+    name: "person"
+ 
+  jailstay:
+    fqn: "ol.jailstay"
+    entitySetName: "MSOJailstay"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
+        column: "SYSID"
+    name: "jailstay"
+
+  charge:
+    fqn: "ol.offense"
+    entitySetName: "MSOCharge"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
```

Comment:
This charge pk is not different for each instance of the charge, is it? If so would prevent de-duplication.

Reply:
Charge PK is unique for every row in this table - in cc_history it is not, there are up to ~9 rows for each charge_pk. I could deduplicate on offense_code instead, which has the following top values:  ```
265/13A/B          10423
A&B                 6898
265/15A/A           5944
266/30/A            5823
209A/7              5041
```

Reply:
The point of having both a charge and a charge event entity type was to de-dupe charges, so we definitely do not want the charge_pk as the primary key here. It should be the pk of the charge event. I'd just concatenate all the columns here for the pk.

Reply:
The columns in this charge entity (excluding CHARGE_PK, since  i'm removing that) have  these properties:

- CHARGE_ORDER is a number which appears to be indicating the number of charges, or the order in which they were received? One is the most common value(~176k rows), followed by two (29k), decreasing in the number of occurrences all the way up to 59. All numbers greater than 22 only have one instance. Including this would probably lead to more specificity than it seems like you are talking about with deduplicating charges.
- CHARGE_TYPE_ID is sparsely populated and only ever takes the values MA (323) and MAP (7) - possibly worth asking Sonya about this one too since it doesn't give much information.
- OFFENSE_DESCRIPTION seems to be associated nearly 1-1 with OFFENSE_CODE (at least based on checking the top few values) - but is more sparsely populated

Based on the above I think it makes most sense to deduplicate on OFFENSE_CODE and i included that in this latest commit